File: QMainDemo/Android/src/io/kanbuki/qt/android/vista/AndroidServiceManager.java
package io.kanbuki.qt.android.vista;

import android.content.Context;
import android.os.Vibrator;
import android.os.BatteryManager;
public class AndroidServiceManager {
    //震动服务
    public static void startVibratorService(Context context){
        Vibrator m_Vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
        long[] pattern = {200, 200};
        m_Vibrator.vibrate(pattern, -1);
    }
    //电池服务 
}

File: QtVista/VistaAndroid/AndroidServiceManager.cpp
#include "AndroidServiceManager.h"

#include <QtCore/private/qandroidextras_p.h>

AndroidServiceManager::AndroidServiceManager(QObject *parent) : QObject{parent}
{

}
QObject* AndroidServiceManager::qmlInstance(QQmlEngine *engine, QJSEngine *scriptEngine){
    Q_UNUSED(engine);
    Q_UNUSED(scriptEngine);
    AndroidServiceManager *androidServiceManager = new AndroidServiceManager();
    return androidServiceManager;
}

void AndroidServiceManager::startVibratorService(){
    QJniObject::callStaticMethod<void>(
                "io/kanbuki/qt/android/vista/AndroidServiceManager",
                "startVibratorService",
                "(Landroid/content/Context;)V",
                 QtAndroidPrivate::context());
}
